### ISSUE TITLE
Confusing implementation of spectral normalization

### ISSUE THREAD

**Issue by daigo0927** [daigo0927/biggan-tensorflow#8]
It is confusing that implementing the spectral normalization layer with `tf.keras.layers.Layer` API.
Using `tf.keras.layers.Wrapper` would be better (like https://github.com/thisisiron/spectral_normalization-tf2/blob/master/sn.py).
